### PR TITLE
added validation for models

### PR DESCRIPTION
**Pull request by jcgrethe**
## Summary
- Added preconditions validations for setter in Book and User models

### CODE REVIEW

**Review by Gianni-Nicolas — APPROVED** [review]

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -148,6 +158,7 @@ public String getIsbn() {
     }
 
     public void setIsbn(String isbn) {
+        Preconditions.checkNotNull(isbn);
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: You could check that the isbn is only have numerical characters.

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -84,14 +85,16 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre);
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/models/Book.java]: The method `checkNotNull(T element)` returns an object of type `T`, I think you should be able to do something like this:

`this.genre = Preconditions.checkNotNull(genre);` which is a bit cleaner :) 

The same goes for all uses of the method (but unfortunately it doesn't apply to the method `checkArgument`)

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Nice job 👍 a few tips and it's ready to go

**Review by Gianni-Nicolas — APPROVED** [review]

**Review by Jimenarosello — APPROVED** [review]
Nice job 😄

**Pull request merged**